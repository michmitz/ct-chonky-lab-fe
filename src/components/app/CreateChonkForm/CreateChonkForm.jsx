import React from 'react';
import { useState } from 'react';
import { createChonk } from '../../../actions/chonkActions';
import { useDispatch } from 'react-redux';

const CreateChonkForm = () => {
  const [name, setName] = useState('');
  const [weight, setWeight] = useState(1);
  const [description, setDescription] = useState('');
  const [imageUrl, setImageUrl] = useState('');
  const dispatch = useDispatch();
  
  const handleChange = ({ target }) => {
    if(target.name === 'name') setName(target.value);
    if(target.name === 'weight') setWeight(target.value);
    if(target.name === 'description') setDescription(target.value);
    if(target.name === 'imageUrl') setImageUrl(target.value);
  };

  const handleSubmit = e => {
    e.preventDefault();
    dispatch(createChonk({
      name,
      weight,
      description,
      imageUrl
    }));
    setName('');
    setWeight('');
    setDescription('');
    setImageUrl('');
  }
  
  return (
    <form onSubmit={handleSubmit}> 
      <h2>Create a new chonk!</h2>
      <input 
        type="text" 
        name="name" 
        value={name} 
        onChange={handleChange} />

      <input 
        type="number" 
        name="weight"
        value={weight}
        onChange={handleChange} />

      <input 
        type="text" 
        name="description"
        value={description}
        onChange={handleChange} />

      <input 
        type="text" 
        name="imageUrl" 
        value={imageUrl}
        onChange={handleChange} />

      <button>Submit Chonk</button>
    </form>
  )
}

export default CreateChonkForm;
