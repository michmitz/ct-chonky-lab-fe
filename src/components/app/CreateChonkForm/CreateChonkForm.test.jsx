import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CreateChonkForm from './CreateChonkForm';
import { Provider } from 'react-redux';
import store from '../../../store';

describe('CreateChonkForm component', () => {
  afterEach(() => cleanup());
  it('renders CreateChonkForm', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <CreateChonkForm />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});