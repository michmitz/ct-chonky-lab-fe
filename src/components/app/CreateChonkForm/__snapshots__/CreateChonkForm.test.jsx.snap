// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateChonkForm component renders CreateChonkForm 1`] = `
<DocumentFragment>
  <form>
    <h2>
      Create a new chonk!
    </h2>
    <input
      name="name"
      type="text"
      value=""
    />
    <input
      name="weight"
      type="number"
      value="1"
    />
    <input
      name="description"
      type="text"
      value=""
    />
    <input
      name="imageUrl"
      type="text"
      value=""
    />
    <button>
      Submit Chonk
    </button>
  </form>
</DocumentFragment>
`;
